Userland/Services/WindowServer/ConnectionFromClient.cpp

```diff
@@ -1298,34 +1304,44 @@ void ConnectionFromClient::set_window_parent_from_client(i32 client_id, i32 pare
 
 Messages::WindowServer::GetWindowRectFromClientResponse ConnectionFromClient::get_window_rect_from_client(i32 client_id, i32 window_id)
 {
-    auto client_connection = from_client_id(client_id);
-    if (!client_connection)
+    auto* client_connection = from_client_id(client_id);
+    if (!client_connection) {
         did_misbehave("GetWindowRectFromClient: Bad client ID");
+        return 0;
```

Comment:
Since you asked about this: What you return has to be a valid way to construct the result of the IPC message.

This IPC message is defined here: https://github.com/SerenityOS/serenity/blob/bf2debdf53fe8246f1b209c87e225a794516e767/Userland/Services/WindowServer/WindowServer.ipc#L181

The return type is `(Gfx::IntRect rect)`, so to be completely correct you want to return an IntRect, wrapped in `{}` (because we actually return a `GetWindowRectFromClientResponse` that *contains* the IntRect). So, something like this:

```suggestion
        return { Gfx::IntRect() };
```

Unfortunately there is a bug where returning something invalid does not give you a compiler error, it just makes the program freeze when it tries to send it.

Reply:
OK, That's what i'm looking for